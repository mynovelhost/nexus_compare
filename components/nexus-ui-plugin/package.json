{
  "name": "nexus-ui-plugin",
  "version": "0.1.0",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "private": true,
  "dependencies": {
    "@fortawesome/fontawesome-svg-core": "=1.2.25",
    "@fortawesome/free-solid-svg-icons": "=5.11.2",
    "@fortawesome/react-fontawesome": "=0.1.4",
    "@sonatype/react-shared-components": "=0.45.11",
    "classnames": "=2.2.6"
  },
  "peerDependencies": {
    "react": "=16.11.0",
    "react-dom": "=16.11.0",
    "xstate": "=4.7.5"
  },
  "devDependencies": {
    "@babel/cli": "=7.6.4",
    "@babel/plugin-proposal-class-properties": "=7.7.0",
    "@babel/plugin-proposal-optional-chaining": "=7.6.0",
    "@testing-library/jest-dom": "=4.2.4",
    "@testing-library/react": "=9.4.0",
    "cpx": "=1.5.0",
    "jest": "=24.8.0",
    "jest-junit": "=10.0.0",
    "node-sass": "=4.13.0",
    "npm-run-all": "=4.1.5",
    "prop-types": "=15.7.2",
    "react": "=16.11.0",
    "react-dom": "=16.11.0",
    "react-test-renderer": "=16.11.0",
    "xstate": "=4.7.5"
  },
  "scripts": {
    "build": "npm-run-all build-all",
    "build-all": "npm-run-all compile-js copy-scss copy-package-json",
    "watch": "npm-run-all copy-scss copy-package-json watch-js",
    "compile-js": "babel src/frontend/src --source-maps inline --out-dir dist",
    "watch-js": "babel src/frontend/src --out-dir dist --watch --verbose",
    "copy-scss": "cpx \"src/frontend/src/**/*.scss\" dist",
    "copy-package-json": "cpx package.json dist",
    "test": "jest --config src/frontend/jest.config.js",
    "test-coverage": "jest --config src/frontend/jest.config.js --coverage",
    "test-watch": "jest --config src/frontend/jest.config.js --watch",
    "test-watch-debug": "node --inspect-brk node_modules/.bin/jest --config src/frontend/jest.config.js --runInBand --watchAll"
  }
}
