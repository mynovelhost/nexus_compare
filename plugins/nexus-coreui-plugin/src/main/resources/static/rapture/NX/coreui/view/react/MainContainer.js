/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
