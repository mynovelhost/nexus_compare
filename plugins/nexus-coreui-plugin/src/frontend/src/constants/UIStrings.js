/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import { UIStrings } from 'nexus-ui-plugin';
import React from 'react';

export default {
  ...UIStrings,

  ANONYMOUS_SETTINGS: {
    MENU: {
      text: 'Anonymous Access',
      description: 'Configure anonymous access to server contents'
    },
    ENABLED_CHECKBOX_LABEL: 'Access:',
    ENABLED_CHECKBOX_DESCRIPTION: 'Allow anonymous users to access the server',
    USERNAME_TEXTFIELD_LABEL: 'Username:',
    REALM_SELECT_LABEL: 'Realm:',
    MESSAGES: {
      LOAD_ERROR: 'An error occurred while loading Anonymous settings, see console for more details',
      SAVE_SUCCESS: 'Anonymous security settings updated',
      SAVE_ERROR: 'An error occurred while updating Anonymous settings, see console for more details'
    }
  },

  LOGGING: {
    MENU: {
      text: 'Logging',
      description: 'Control logging levels',
      edit: (name) => `Edit ${name}`
    },

    CREATE_BUTTON: 'Create Logger',
    RESET_ALL_BUTTON: 'Reset to Default Levels',
    RESET_BUTTON: 'Reset to Default Level',

    FILTER_PLACEHOLDER: 'Filter by Logger Name',

    NAME_LABEL: 'Logger Name',
    LEVEL_LABEL: 'Logger Level',

    MESSAGES: {
      SAVE_ERROR: 'An error occurred while saving the logger',
      RESETTING: 'Resetting to Default Level...',
      RESET_ERROR: 'An error occurred while resetting all loggers, see console for more details'
    },

    CONFIRM_UPDATE: {
      TITLE: 'Confirm Update?',
      MESSAGE: ({name, level}) => `Logger ${name} is already configured. Would you like to update its level to "${level}"?`,
      CONFIRM_BUTTON: 'Update Logger Level'
    },

    CONFIRM_RESET_ALL: {
      TITLE: 'Confirm Reset?',
      MESSAGE: 'Reset all loggers to their default levels?',
      CONFIRM_BUTTON: 'Reset All Loggers'
    },

    CONFIRM_RESET: {
      TITLE: 'Confirm Reset?',
      MESSAGE: 'Reset this logger to its default level? If this is a custom logger it will be removed.',
      CONFIRM_BUTTON: 'Reset Logger'
    }
  },

  LOG_VIEWER: {
    MENU: {
      text: 'Log Viewer',
      description: 'View the current log contents'
    },
    REFRESH: {
      TEXT: 'Refresh Interval',
      MANUAL_ITEM: 'Manual',
      TWENTY_SECONDS_ITEM: 'Every 20 seconds',
      MINUTE_ITEM: 'Every minute',
      TWO_MINUTES_ITEM: 'Every 2 minutes',
      FIVE_MINUTES_ITEM: 'Every 5 minutes'
    },
    SIZE: {
      LAST25KB_ITEM: 'Last 25KB',
      LAST50KB_ITEM: 'Last 50KB',
      LAST100KB_ITEM: 'Last 100KB'
    },
    DOWNLOAD: 'Download',
    INSERT_MARK: 'Insert Mark',
    MARK_PLACEHOLDER: 'Marker to insert into log'
  },

  SYSTEM_INFORMATION: {
    MENU: {
      text: 'System Information'
    },
    ACTIONS: {
      download: 'Download as JSON'
    }
  },

  SUPPORT_REQUEST: {
    MENU: {
      text: 'Support Request',
      description: 'Submit a support request to Sonatype'
    },
    ACTIONS: {
      submitRequest: 'Submit Request'
    },
    DESCRIPTION: `Please include a complete description of your problem and steps to allow us to reproduce the problem (if available).`,
    ATTACH_SUPPORT_ZIP: `Attaching a support ZIP to your request will help our engineers give you a faster response.`
  },

  USER_ACCOUNT: {
    MENU: {
      text: 'Account',
      description: 'Manage your account',
    },
    ACTIONS: {
      changePassword: 'Change password',
      discardChangePassword: 'Discard password change',
    },
    MESSAGES: {
      LOAD_ERROR: 'An error occurred while loading User Account, see console for more details',
      UPDATE_SUCCESS: 'User account settings updated',
      UPDATE_ERROR: 'An error occurred while updating user account settings',
      PASSWORD_NO_MATCH_ERROR: 'Passwords do not match',
      PASSWORD_MUST_DIFFER_ERROR: 'New password must be different',
    },
    ID_FIELD_LABEL: 'Username',
    FIRST_FIELD_LABEL: 'First Name',
    LAST_FIELD_LABEL: 'Last Name',
    EMAIL_FIELD_LABEL: 'Email',
    PASSWORD_CURRENT_FIELD_LABEL: 'Current Password',
    PASSWORD_NEW_FIELD_LABEL: 'New Password',
    PASSWORD_NEW_CONFIRM_FIELD_LABEL: 'New Password (Confirm)',
  },

  NUGET_API_KEY: {
    TITLE: 'NuGet API Key',
    DESCRIPTION: 'Configure credentials for NuGet repositories',
    INSTRUCTIONS: 'A new API Key will be created the first time it is accessed. Resetting your API Key will invalidate the current key.',
    AUTH_INSTRUCTIONS: 'Accessing NuGet API Key requires validation of your credentials.',
    AUTH_ERROR: 'Authentication Failure',
    CLOSE: 'Close',
    ACCESS: {
      BUTTON: 'Access API Key',
      HELP_TEXT: 'Accessing NuGet API Key requires validation of your credentials.',
      ERROR: 'Failed to access API Key'
    },
    RESET: {
      BUTTON: 'Reset API Key',
      HELP_TEXT: 'Resetting NuGet API Key requires validation of your credentials.',
      ERROR: 'Failed to Reset API Key'
    },
    DETAILS: {
      MAIN: 'Your NuGet API Key enables pushing packages using nuget.exe.' ,
      WARNING:'Keep this key secret!' ,
      API_KEY_TEXT: 'Your NuGet API Key is:',
      REGISTER_TEXT: 'You can register this key for a given repository with the following command:',
      REGISTER_COMMAND: 'nuget setapikey {0} -source {1}',
      AUTO_CLOSE: 'This window will automatically close after one minute.'
    }
  },

  METRIC_HEALTH: {
    NAME_HEADER: 'Name',
    MESSAGE_HEADER: 'Message',
    ERROR_HEADER: 'Error',
  },

  SUPPORT_ZIP: {
    MENU: {
      text: 'Support ZIP',
      description: 'Creates a ZIP file containing useful support information about your server'
    },
    DESCRIPTION: '<p>No information will be sent to Sonatype when creating the support ZIP file.</p>' +
        '<p>Support ZIP creation may take a few minutes to complete.</p>',
    REPORT_LABEL: 'System information report',
    DUMP_LABEL: 'JVM thread-dump',
    CONFIGURATION_LABEL: 'Configuration files',
    SECURITY_LABEL: 'Security configuration files',
    LOGFILES_LABEL: 'Log files',
    TASKLOGFILES_LABEL: 'Task log files',
    AUDITLOGFILES_LABEL: 'Audit log files',
    METRICS_LABEL: 'System and component metrics',
    JMX_LABEL: 'JMX information',
    LIMITFILESIZES_LABEL: 'Limit files in the ZIP archive to 30 MB apiece',
    LIMITZIPSIZE_LABEL: 'Limit the ZIP archive to 20 MB',
    CREATED_TITLE: 'Support ZIP Created',
    CREATED_DESCRIPTION: 'Support ZIP has been created.<br/>You can reference this file on the filesystem or download the file from your browser.',
    CREATED_NODEID_LABEL: 'Node:',
    CREATED_NAME_LABEL: 'Name:',
    CREATED_SIZE_LABEL: 'Size:',
    CREATED_PATH_LABEL: 'Path:',
    CREATED_DOWNLOAD_BUTTON: 'Download',
    AUTHENTICATE_TEXT: 'Downloading support ZIP requires validation of your credentials.'
  }
};
