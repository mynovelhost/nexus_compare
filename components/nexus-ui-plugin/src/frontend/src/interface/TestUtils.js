/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {act} from 'react-dom/test-utils';
import {render} from '@testing-library/react';

/**
 * @since 3.24
 */
export default class Utils {
  static render(view, extraSelectors) {
    const selectors = render(view);
    const {queryByText} = selectors;
    return {
      ...selectors,
      loadingMask: () => queryByText('Loadingâ€¦'),
      ...extraSelectors(selectors)};
  }
}
