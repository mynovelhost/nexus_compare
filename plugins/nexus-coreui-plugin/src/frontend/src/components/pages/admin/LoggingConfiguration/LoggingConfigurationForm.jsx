/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import {useMachine} from '@xstate/react';

import {
  Button,
  FieldWrapper,
  NxErrorAlert,
  NxLoadWrapper,
  NxSubmitMask,
  Section,
  SectionFooter,
  Select,
  Textfield
} from "nexus-ui-plugin";

import LoggingConfigurationFormMachine from "./LoggingConfigurationFormMachine";

import UIStrings from "../../../../constants/UIStrings";

export default function LoggingConfigurationForm({itemId, onDone}) {
  const [current, send] = useMachine(LoggingConfigurationFormMachine, {
    context: {
      pristineData: {
        name: itemId
      }
    },

    actions: {
      onDone
    },

    devTools: true
  });
  const {isPristine, isInvalid, pristineData, data, loadError, saveError} = current.context;
  const isLoading = current.matches('loading');
  const isSaving = current.matches('saving');
  const isResetting = current.matches('resetting');

  function update(event) {
    send('UPDATE', {data: {[event.target.name]: event.target.value}});
  }

  function save(event) {
    event.preventDefault();
    send('SAVE');
  }

  function handleEnter(event) {
    if (event.key === 'Enter') {
      save(event);
    }
  }

  function cancel() {
    send('CANCEL');
  }

  function reset() {
    send('RESET');
  }

  return <Section className="nxrm-logging-configuration-form" onKeyPress={handleEnter}>
    <NxLoadWrapper loading={isLoading} error={loadError ? `${loadError}` : null}>
      {saveError && <NxErrorAlert>{UIStrings.LOGGING.MESSAGES.SAVE_ERROR} {saveError}</NxErrorAlert>}
      {isSaving && <NxSubmitMask message={UIStrings.SAVING}/>}
      {isResetting && <NxSubmitMask message={UIStrings.LOGGING.MESSAGES.RESETTING}/>}

      <FieldWrapper labelText={UIStrings.LOGGING.NAME_LABEL}>
        <Textfield isRequired disabled={pristineData.name} name="name" value={data.name} onChange={update}/>
      </FieldWrapper>
      <FieldWrapper labelText={UIStrings.LOGGING.LEVEL_LABEL}>
        <Select name="level" value={data.level} onChange={update}>
          {['OFF', 'ERROR', 'WARN', 'INFO', 'DEBUG', 'TRACE'].map(logLevel =>
              <option key={logLevel} value={logLevel}>{logLevel}</option>
          )}
        </Select>
      </FieldWrapper>

      <SectionFooter>
        <Button variant="primary" disabled={isPristine || isInvalid} onClick={save} type="submit">
          {UIStrings.SETTINGS.SAVE_BUTTON_LABEL}
        </Button>
        <Button onClick={cancel}>{UIStrings.SETTINGS.CANCEL_BUTTON_LABEL}</Button>
        {itemId && <Button variant="error" onClick={reset}>{UIStrings.LOGGING.RESET_BUTTON}</Button>}
      </SectionFooter>
    </NxLoadWrapper>
  </Section>;
}
