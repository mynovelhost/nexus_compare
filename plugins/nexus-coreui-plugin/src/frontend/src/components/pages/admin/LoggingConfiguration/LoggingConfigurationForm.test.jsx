/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import { act } from 'react-dom/test-utils';
import {fireEvent, render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestUtils from 'nexus-ui-plugin/src/frontend/src/interface/TestUtils';
import axios from 'axios';
import {ExtJS} from 'nexus-ui-plugin';

import LoggingConfigurationForm from './LoggingConfigurationForm';

import UIStrings from "../../../../constants/UIStrings";

jest.mock('axios', () => ({
  ...jest.requireActual('axios'), // Use most functions from actual axios
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn()
}));

jest.mock('nexus-ui-plugin', () => ({
  ...jest.requireActual('nexus-ui-plugin'),
  ExtJS: {
    setDirtyStatus: jest.fn(),
    requestConfirmation: jest.fn()
  }
}));

describe('LoggingConfigurationForm', function() {
  const CONFIRM = Promise.resolve();
  const OVERRIDDEN_LOGGER = Promise.resolve({data: {override: true}});
  const onDone = jest.fn();

  const renderEditView = async () => {
    return renderView(<LoggingConfigurationForm itemId="ROOT" onDone={onDone}/>);
  };

  const renderCreateView = async () => {
    return renderView(<LoggingConfigurationForm onDone={onDone} />);
  };

  function renderView(view) {
    return TestUtils.render(view, ({queryByLabelText, queryByText}) => ({
      name: () => queryByLabelText(UIStrings.LOGGING.NAME_LABEL),
      level: () => queryByLabelText(UIStrings.LOGGING.LEVEL_LABEL),
      saveMask: () => queryByText(UIStrings.SAVING),
      saveButton: () => queryByText(UIStrings.SETTINGS.SAVE_BUTTON_LABEL),
      cancelButton: () => queryByText(UIStrings.SETTINGS.CANCEL_BUTTON_LABEL),
      resetMask: () => queryByText(UIStrings.LOGGING.MESSAGES.RESETTING),
      resetButton: () => queryByText(UIStrings.LOGGING.RESET_BUTTON)
    }));
  }

  const changeFieldAndAssertValue = async (fieldSelector, value) => {
    fireEvent.change(fieldSelector(), {target: {value: value}});
    await wait(() => expect(fieldSelector()).toHaveValue(value));
  };

  it('renders the resolved data', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: {name: 'ROOT', level: 'INFO'}
    }));

    const {container, loadingMask, name, level, saveButton} = await renderEditView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(name()).toHaveValue('ROOT');
    expect(level()).toHaveValue('INFO');
    expect(saveButton()).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  it('renders a loading spinner', async function() {
    axios.get.mockReturnValue(new Promise(() => {}));

    const {container, loadingMask} = await renderEditView();

    expect(loadingMask()).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders an error message', async function() {
    axios.get.mockReturnValue(Promise.reject({message: 'Error'}));

    const {container, loadingMask} = await renderEditView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(container.querySelector('.nx-alert--error')).toHaveTextContent('Error');
    expect(container).toMatchSnapshot();
  });

  it('requires the name field when creating a new logging configuration', async function() {
    const {container, loadingMask, name, level, saveButton} = await renderCreateView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(container).toMatchSnapshot();

    expect(saveButton()).toBeDisabled();
    expect(name()).toBeRequired();
    expect(level()).toHaveValue('INFO');

    await changeFieldAndAssertValue(name, 'name');

    expect(saveButton()).toBeEnabled();
  });

  it('fires onDone when cancelled', async function() {
    const {loadingMask, cancelButton} = await renderCreateView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    fireEvent.click(cancelButton());

    expect(onDone).toBeCalled();
  });

  it('requests confirmation when the logger is overridden and saves when requested', async function() {
    const {loadingMask, name, level, saveMask, saveButton} = await renderCreateView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    axios.put.mockReturnValue(Promise.resolve());

    await changeFieldAndAssertValue(name, 'name');
    await changeFieldAndAssertValue(level, 'DEBUG');
    expect(saveButton()).toBeEnabled();

    axios.get.mockReturnValue(OVERRIDDEN_LOGGER);
    ExtJS.requestConfirmation.mockReturnValue(CONFIRM);
    fireEvent.click(saveButton());

    await wait(() => expect(saveMask()).toBeInTheDocument());

    expect(axios.put).toBeCalledWith(
        '/service/rest/internal/ui/loggingConfiguration/name',
        {name: 'name', level: 'DEBUG'}
    );
    expect(onDone).toBeCalled();
  });

  it('saves', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: {name: 'ROOT', level: 'INFO'}
    }));

    const {loadingMask, level, saveMask, saveButton} = await renderEditView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(ExtJS.setDirtyStatus).toHaveBeenLastCalledWith('EditLoggingConfiguration', false);

    await changeFieldAndAssertValue(level, 'DEBUG');

    expect(ExtJS.setDirtyStatus).toHaveBeenLastCalledWith('EditLoggingConfiguration', true);

    fireEvent.click(saveButton());

    await wait(() => expect(saveMask()).toBeInTheDocument());

    expect(axios.put).toHaveBeenLastCalledWith(
        '/service/rest/internal/ui/loggingConfiguration/ROOT',
        {name: 'ROOT', level: 'DEBUG'}
    );
    expect(ExtJS.setDirtyStatus).toHaveBeenLastCalledWith('EditLoggingConfiguration', false);
  });

  it('resets an edited logger', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: {name: 'test', level: 'ERROR'}
    }));

    const {loadingMask, resetButton, resetMask} = await renderEditView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    ExtJS.requestConfirmation.mockReturnValue(CONFIRM);
    axios.post.mockReturnValue(CONFIRM);
    fireEvent.click(resetButton());

    await wait(() => expect(resetMask()).toBeInTheDocument());

    expect(axios.post).toBeCalledWith('/service/rest/internal/ui/loggingConfiguration/test/reset');
    expect(onDone).toBeCalled();
  });
});
