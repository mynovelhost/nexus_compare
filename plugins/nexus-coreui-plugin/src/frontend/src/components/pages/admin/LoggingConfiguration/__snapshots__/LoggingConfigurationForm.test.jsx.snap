// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggingConfigurationForm renders a loading spinner 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-form"
  >
    <div
      class="nx-loading-spinner"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-notch fa-w-16 fa-spin nx-loading-spinner__icon nx-icon"
        data-icon="circle-notch"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
          fill="currentColor"
        />
      </svg>
      <span>
        Loadingâ€¦
      </span>
    </div>
  </div>
</div>
`;

exports[`LoggingConfigurationForm renders an error message 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-form"
  >
    <div
      class="nx-alert nx-alert--error"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times-circle fa-w-16 nx-icon"
        data-icon="times-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
          fill="currentColor"
        />
      </svg>
      <span>
        An error occurred loading data.
      </span>
       
      <span>
        Error
      </span>
    </div>
  </div>
</div>
`;

exports[`LoggingConfigurationForm renders the resolved data 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-form"
  >
    <div
      class="field-wrapper"
    >
      <label
        class="field-wrapper-label"
        for="name"
      >
        Logger Name
      </label>
      <input
        class="nxrm-textfield"
        disabled=""
        id="name"
        name="name"
        required=""
        type="text"
        value="ROOT"
      />
    </div>
    <div
      class="field-wrapper"
    >
      <label
        class="field-wrapper-label"
        for="level"
      >
        Logger Level
      </label>
      <select
        class="nxrm-select"
        id="level"
        name="level"
      >
        <option
          value="OFF"
        >
          OFF
        </option>
        <option
          value="ERROR"
        >
          ERROR
        </option>
        <option
          value="WARN"
        >
          WARN
        </option>
        <option
          value="INFO"
        >
          INFO
        </option>
        <option
          value="DEBUG"
        >
          DEBUG
        </option>
        <option
          value="TRACE"
        >
          TRACE
        </option>
      </select>
    </div>
    <div
      class="nxrm-section-footer nx-tile-footer"
    >
      <button
        class="nx-btn nx-btn--primary"
        disabled=""
        type="submit"
      >
        Save
      </button>
      <button
        class="nx-btn"
        type="input"
      >
        Cancel
      </button>
      <button
        class="nx-btn nx-btn--error"
        type="input"
      >
        Reset to Default Level
      </button>
    </div>
  </div>
</div>
`;

exports[`LoggingConfigurationForm requires the name field when creating a new logging configuration 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-form"
  >
    <div
      class="field-wrapper"
    >
      <label
        class="field-wrapper-label"
        for="name"
      >
        Logger Name
      </label>
      <input
        class="nxrm-textfield missing-required-value"
        id="name"
        name="name"
        required=""
        type="text"
        value=""
      />
      <span
        class="nxrm-error-message"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation-circle fa-w-16 nx-icon"
          data-icon="exclamation-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
        <span
          class="nxrm-error-message-text"
          tabindex="-1"
        >
           
          This field is required
           
        </span>
      </span>
    </div>
    <div
      class="field-wrapper"
    >
      <label
        class="field-wrapper-label"
        for="level"
      >
        Logger Level
      </label>
      <select
        class="nxrm-select"
        id="level"
        name="level"
      >
        <option
          value="OFF"
        >
          OFF
        </option>
        <option
          value="ERROR"
        >
          ERROR
        </option>
        <option
          value="WARN"
        >
          WARN
        </option>
        <option
          value="INFO"
        >
          INFO
        </option>
        <option
          value="DEBUG"
        >
          DEBUG
        </option>
        <option
          value="TRACE"
        >
          TRACE
        </option>
      </select>
    </div>
    <div
      class="nxrm-section-footer nx-tile-footer"
    >
      <button
        class="nx-btn nx-btn--primary"
        disabled=""
        type="submit"
      >
        Save
      </button>
      <button
        class="nx-btn"
        type="input"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
